File: mobile/ios/FUN/FUN/Models/WatchHistory.swift
/**
 * FUN App - Watch History Models
 */

import Foundation

struct WatchHistoryEntry: Codable, Identifiable {
    let id: String // episodeId
    let episodeId: String
    let seriesId: String
    let seasonNumber: Int
    let episodeNumber: Int
    var progress: Double // seconds watched
    let duration: Double
    var completed: Bool
    var watchedAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id = "episodeId"
        case episodeId, seriesId, seasonNumber, episodeNumber
        case progress, duration, completed, watchedAt
    }
}

struct SeriesProgress: Codable {
    let seriesId: String
    var lastWatchedEpisodeId: String
    var lastWatchedSeasonNumber: Int
    var lastWatchedEpisodeNumber: Int
    var completedSeasons: [Int]
    var totalWatchTime: Double
}

struct ContinueWatchingInfo {
    let episodeId: String
    let seriesId: String
    let seasonNumber: Int
    let episodeNumber: Int
    let progress: Double
    let progressPercent: Double
}

File: mobile/android/FUN/app/src/main/java/com/fun/app/data/models/WatchHistory.kt
/**
 * FUN App - Watch History Models
 */

package com.fun.app.data.models

import com.squareup.moshi.JsonClass
import java.util.Date

@JsonClass(generateAdapter = true)
data class WatchHistoryEntry(
    val episodeId: String,
    val seriesId: String,
    val seasonNumber: Int,
    val episodeNumber: Int,
    var progress: Double, // seconds watched
    val duration: Double,
    var completed: Boolean,
    var watchedAt: Date
)

@JsonClass(generateAdapter = true)
data class SeriesProgress(
    val seriesId: String,
    var lastWatchedEpisodeId: String,
    var lastWatchedSeasonNumber: Int,
    var lastWatchedEpisodeNumber: Int,
    var completedSeasons: MutableList<Int>,
    var totalWatchTime: Double
)

data class ContinueWatchingInfo(
    val episodeId: String,
    val seriesId: String,
    val seasonNumber: Int,
    val episodeNumber: Int,
    val progress: Double,
    val progressPercent: Double
)
